test/src/lib/update-lock.spec.ts

```diff
@@ -108,40 +112,101 @@ describe('lib/update-lock', () => {
 		});
 	});
 
+	describe('cleanupLocks', () => {
+		it('should cleanup all leftover Supervisor-created lockfiles', async () => {
+			const BASE_PATH = '/mnt/root/tmp/balena-supervisor/services';
+
+			// Setup fake filesystem & stubs
+			mockFs({
+				[BASE_PATH]: {
+					'1234567': {
+						one: {
+							'updates.lock': mockFs.file({ uid: lockfile.NOBODY_UID }),
+						},
+						two: {
+							'updates.lock': mockFs.file({ uid: 0 }),
+						},
+					},
+					'7654321': {
+						three: {
+							'updates.lock': mockFs.file({ uid: lockfile.NOBODY_UID }),
+						},
+						four: {
+							'updates.lock': mockFs.file({ uid: 0 }),
+						},
+					},
+				},
+			});
+			stub(fsUtils, 'exec').resolves({
+				stdout: `${BASE_PATH}/1234567/one/updates.lock\n${BASE_PATH}/7654321/three/updates.lock\n`,
+				stderr: '',
+			});
+
+			await updateLock.cleanupLocks();
+
+			// These directories only contain lockfiles with `nobody` UID and should be removed
+			expect(
+				await fs.readdir(path.join(BASE_PATH, '1234567', 'one')),
+			).to.have.length(0);
+			expect(
+				await fs.readdir(path.join(BASE_PATH, '7654321', 'three')),
+			).to.have.length(0);
+
+			// These directories contain lockfiles with `root` UIDs and should remain
+			expect(
+				await fs.readdir(path.join(BASE_PATH, '1234567', 'two')),
+			).to.have.length(1);
+			expect(
+				await fs.readdir(path.join(BASE_PATH, '7654321', 'four')),
+			).to.have.length(1);
+
+			mockFs.restore();
+			(fsUtils.exec as SinonStub).restore();
+		});
+	});
+
 	describe('Lock/dispose functionality', () => {
-		const lockFile = updateLock.__get__('lockFile');
```

Comment:
Can we re-do these tests without using rewire?

Reply:
Depending on the outcome of our discussion on `lockfileErrHandler` above, we could remove that, but I don't think `dispose` could be modified to not use rewire unless the method is moved to `lib/lockfile.ts` as well. I opted to keep `dispose` in update-lock.ts as it has more to do with the Supervisor-specific update-lock functionality than the generic lockfile lib.

Reply:
Updated tests have removed the dependency on `rewire`.